client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
@cyli I'd go the other way. Add a flag that we can check to determine we have performed an update. On the basis this won't be the only time we want to perform multiple ops against a loaded repo, having a consistent way for any method that _might_ need to load the repo to check if it's already been loaded seems like it would be useful.

Maybe even have the flag as a `time.Time` and allow `Update` to decide if a sufficient amount of time has passed that it should actually run an update again.